src/librustc/hir/map/mod.rs

```diff
@@ -814,6 +948,14 @@ impl<'hir> Map<'hir> {
         }
     }
 
+    // FIXME(@ljedrz): replace the NodeId variant
+    pub fn expect_item_by_hir_id(&self, id: HirId) -> &'hir Item {
+        match self.find_by_hir_id(id) { // read recorded by `find`
```

Comment:
This should just convert the `HirId` to a `NodeId` and call the regular `expect_item` method.